lxd/instance/drivers/driver_qemu.go

```diff
@@ -2443,12 +2443,20 @@ func (d *qemu) generateQemuConfigFile(mountInfo *storagePools.MountInfo, busName
 
 	// s390x doesn't really have USB.
 	if d.architecture != osarch.ARCH_64BIT_S390_BIG_ENDIAN {
+		// Record the number of USB devices.
+		totalUSBdevs := 0
+
+		for _, runConf := range devConfs {
+			totalUSBdevs += len(runConf.USBDevice)
+		}
+
 		devBus, devAddr, multi = bus.allocate(busFunctionGroupGeneric)
 		err = qemuUSB.Execute(sb, map[string]interface{}{
 			"bus":           bus.name,
 			"devBus":        devBus,
 			"devAddr":       devAddr,
 			"multifunction": multi,
+			"ports":         totalUSBdevs + 4,
```

Comment:
OK can we make this 4 a constant and use it here and with the pcie hotplugging logic so we know why its there.